pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -39,92 +40,132 @@ import (
-		case err == nil && a != nil && a.Status.ReadyReplicas == 1:
-			if unschedulableNodes == nil {
-				ns, err := c.KubeClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
-					FieldSelector: "spec.unschedulable=true",
-				})
-				if err != nil {
-					errs = append(errs, fmt.Errorf("could not get the list of nodes, err: %w", err))
-				}
-				unschedulableNodes = sets.Set[string]{}
-				// maintain the list of nodes which are unschedulable
-				for _, n := range ns.Items {
-					unschedulableNodes.Insert(n.Name)
-				}
+		}
+		if unschedulableNodes.Len() > 0 {
+			// get the pod created for a given StatefulSet, pod is assigned ordinal of 0 with the replicas set to 1
+			p, err := c.KubeClientSet.CoreV1().Pods(pr.Namespace).Get(ctx, a.Name+"-0", metav1.GetOptions{})
+			// ignore instead of failing if the affinity assistant pod was not found
+			if err != nil && !apierrors.IsNotFound(err) {
+				errs = append(errs, fmt.Errorf("could not get the affinity assistant pod for StatefulSet %s: %w", a.Name, err))
 			}
-			if unschedulableNodes.Len() > 0 {
-				// get the pod created for a given StatefulSet, pod is assigned ordinal of 0 with the replicas set to 1
-				p, err := c.KubeClientSet.CoreV1().Pods(pr.Namespace).Get(ctx, a.Name+"-0", metav1.GetOptions{})
-				// ignore instead of failing if the affinity assistant pod was not found
-				if err != nil && !apierrors.IsNotFound(err) {
-					errs = append(errs, fmt.Errorf("could not get the affinity assistant pod for StatefulSet %s: %w", a.Name, err))
-				}
-				// check the node which hosts the affinity assistant pod if it is unschedulable or cordoned
-				if p != nil && unschedulableNodes.Has(p.Spec.NodeName) {
-					// if the node is unschedulable, delete the affinity assistant pod such that a StatefulSet can recreate the same pod on a different node
-					err = c.KubeClientSet.CoreV1().Pods(p.Namespace).Delete(ctx, p.Name, metav1.DeleteOptions{})
-					if err != nil {
-						errs = append(errs, fmt.Errorf("error deleting affinity assistant pod %s in ns %s: %w", p.Name, p.Namespace, err))
-					}
+			// check the node which hosts the affinity assistant pod if it is unschedulable or cordoned
+			if p != nil && unschedulableNodes.Has(p.Spec.NodeName) {
+				// if the node is unschedulable, delete the affinity assistant pod such that a StatefulSet can recreate the same pod on a different node
+				err = c.KubeClientSet.CoreV1().Pods(p.Namespace).Delete(ctx, p.Name, metav1.DeleteOptions{})
+				if err != nil {
+					errs = append(errs, fmt.Errorf("error deleting affinity assistant pod %s in ns %s: %w", p.Name, p.Namespace, err))
 				}
 			}
-		case err != nil:
-			errs = append(errs, fmt.Errorf("failed to retrieve StatefulSet %s: %w", affinityAssistantName, err))
 		}
+	case err != nil:
+		errs = append(errs, fmt.Errorf("failed to retrieve StatefulSet %s: %w", affinityAssistantName, err))
 	}
-	return errorutils.NewAggregate(errs)
+
+	return errs
```

Comment:
do we need to use the `errorutils.NewAggregate(errs)` from the original code? It seems to be used to filter out `nil` from the errs

Reply:
It looks like `NewAggregate()` also formats multiple errors when printing out to help readability, for example:

```
	err1 := fmt.Errorf("first error")
	err2 := fmt.Errorf("second error")

	errs := []error{err1, err2}
	aggregateErr := errorutils.NewAggregate(errs)

	fmt.Printf("error is: %v", errs)  // error is: [first error second error]
	fmt.Printf("error is: %v", aggregateErr)   // error is: [first error, second error]
```

Reply:
Thanks! I mean the `NewAggregate` is not used in this PR's change, do we want to maintain it? But I think this is a nit.

Reply:
Ohh sorry I missed it, let me change it back...

Reply:
Oh sorry, I was wrong. 😂 
I think we can resolve this

Reply:
No worried! ❤️